app.py

```diff
@@ -8,6 +10,64 @@
 app.config['MYSQL_HOST'] = 'localhost'
 mysql = MySQL(app)
 
+# FOR LOCAL SERVER TESTING ONLY
+# Necessary for local server as Flask Dance usually requires https
+import os
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
+os.environ['OAUTHLIB_RELAX_TOKEN_SCOPE'] = '1'
+
+# OAUTH CONFIG AND SETUP
+# ensure ./app.conf exists with client id & secret, and secret key
+app.config.from_pyfile('./app.conf')
+bp_google = make_google_blueprint(
+    client_id = app.config['GOOGLE_CLIENT_ID'],
+    client_secret = app.config['GOOGLE_CLIENT_SECRET'],
+    scope = ["profile", "email"])
+app.register_blueprint(bp_google, url_prefix="/login")
+
+@oauth_authorized.connect
+def logged_in(blueprint, token):
+    if not token:
+        flash("Log in failed.", category="error")
+        return False
+
+    # Google-specific API getter
+    resp = blueprint.session.get("/oauth2/v2/userinfo")
+    
+    if not resp.ok:
+        flash("Failed to get user data.", category="error")
+        return False
+
+    user_info = resp.json()
+    # need to absolutely make sure email is unique per provider,
+    # otherwise, need to use a different unique id
+    social_id = "google." + user_info["email"]
+
+    # add user to database if not found
+    check_query = 'select * from user_profiles where social_id="{}"'
+    cursor = mysql.connection.cursor()
+    cursor.execute("use codestroke")
+    cursor.execute(check_query.format(social_id))
+    result = cursor.fetchall()
+    if not result:
+        email = user_info["email"]
+        name = user_info["name"]
+        insert_query = """insert into user_profiles (social_id, name, email)
+                          values ('{0}', '{1}', '{2}')"""
+        cursor.execute(insert_query.format(social_id, name, email))
```

Comment:
Using .format here can make this query vulnerable to SQL injection attacks. Instead it is better to use the %s specifier instead of the {}'s, and pass a the tuple of values as a 2nd argument to cursor.execute. see add_patient function for an example of that.

Reply:
I've done this, but I think this part should be fairly safe against injection attacks as the strings are derived directly from the Google API responses (and not user input), which already must have certain requirements satisfied (e.g. no spaces in email and therefore no spaces in social ID, no special characters).